docs/gsb_api.md

```diff
@@ -0,0 +1,131 @@
+    App ->> Backend: GET gsb-api/v1/services/<br>L3B1YmxpYy9nZnRwL215YXBwCg== <br> Headers: { Upgrade: websocket }
+    Backend -->> App: HTTP 101
+
+    par WebSocket connection lifespan
+        App ->> Backend: WS Event: Open
+        loop Multiple GSB message calls
+            Service ->>+ Backend: GSB: GetChunk <br>public/gftp/myapp
+            Backend ->> App: WS Message: gsbRequest<br> { id: '<br>L3B1YmxpYy9nZnRwL215YXBwCg==', <br>component: 'GetMetadata', <br>payload: { .. } }
+            App -->> Backend: WS Message: gsbResponse<br> { id: '<br>L3B1YmxpYy9nZnRwL215YXBwCg==', payload: { .. } }
+            Backend -->>- Service : GSB: GftpMetadata
+        end
+        App ->> Backend: DELETE /gsb-api/v1/services/<br>L3B1YmxpYy9nZnRwL215YXBwCg==
+        Backend -->> App: WS Event: Close<br> CloseCode: Normal
+    end
+
+    Backend -->>- App: HTTP 200
+```
+
+#### Early GSB request scenario
+
+```mermaid
+sequenceDiagram
+    participant App as WebSocket Client
+    participant Backend as Backend Yagna
+    participant Service as Golem Service
+
+    App ->>+ Backend: POST /gsb-api/v1/services
+    Backend -->> App: HTTP 201
+    Service ->>+ Backend: GSB: GetChunk
+    Note right of Backend: Buffered GSB request
+    App ->> Backend: GET gsb-api/v1/services/<br>L3B1YmxpYy9nZnRwL215YXBwCg==
+    Backend -->> App: HTTP 101
+    par WebSocket connection lifespan
+        App ->> Backend: WS Event: Open
+        Backend ->> App: WS Message: gsbRequest
+        App -->> Backend: WS Message: gsbResponse
+        Backend -->>- Service : GSB: GftpMetadata
+        App ->> Backend: DELETE /gsb-api/v1/services/<br>L3B1YmxpYy9nZnRwL215YXBwCg==
+        Backend -->> App: WS Event: Close<br> CloseCode: Normal
+    end
+    Backend -->>- App: HTTP 200
+```
+
+### Some other scenarios (simplified)
+
+```mermaid
+sequenceDiagram
+    participant App as WebSocket Client
+    participant Backend as Backend Yagna
+    participant Service as Golem Service
+
+    App ->>+ Backend: POST /gsb-api/v1/services
+    Backend -->> App: HTTP 201
+    App ->> Backend: GET gsb-api/v1/services/<br>L3B1YmxpYy9nZnRwL215YXBwCg==
+    Backend -->> App: HTTP 101
+
+    par 1st WS connection lifespan
+        App -->> Backend: WS Event: Close<br>CloseCode: Abnormal
+    end
+
+    Service ->>+ Backend: GSB: GetMetadata
+    Note right of Backend: Buffering GSB request<br>when WS disconnected
+    App ->> Backend: GET gsb-api/v1/services/<br>L3B1YmxpYy9nZnRwL215YXBwCg==
+    Backend -->> App: HTTP 101
+
+    par 2nd WS connection lifespan
+        App ->> Backend: WS Event: Open
+        Backend ->> App: WS Message: gsbRequest
+        App -->> Backend: WS Message: gsbResponse
+        Backend -->>- Service : GSB: GftpMetadata
+        Service ->>+ Backend: GSB: GetChunk
+        Backend ->> App: WS Message: gsbRequest
+        App -->> Backend: WS Event: Close<br>CloseCode: Abnormal
+    end
+
+    Note right of Backend: Canceling GSB request<br> on WS Close
```

Comment:
Decided to cancel pending GSB requests on WS close instead of buffering them because I assume this Close might have been caused by this GSB requests, so resending it could create a crash loop.